trino/auth.py

```diff
@@ -168,14 +176,24 @@ class _OAuth2TokenBearer(AuthBase):
     MAX_OAUTH_ATTEMPTS = 5
     _BEARER_PREFIX = re.compile(r"bearer", flags=re.IGNORECASE)
 
-    def __init__(self, http_session, redirect_auth_url_handler=handle_redirect_auth_url):
+    def __init__(self, redirect_auth_url_handler=handle_redirect_auth_url):
         self._redirect_auth_url = redirect_auth_url_handler
-        self._thread_local = threading.local()
-        http_session.hooks['response'].append(self._authenticate)
+        try:
+            self._keyring = importlib.import_module("keyring")
+        except ImportError:
+            self._keyring = None
+            print("Warning: keyring module not found. OAuth2 token will not be stored in keyring.")
```

Comment:
Will it be printed for all clients which are not using `keyring`?

Reply:
So as the `_OAuth2TokenBearer` is instantiated as part of the __init__ of the `OAuth2Authentication` class, it will be shown only once. I thought it's a good way to remind the user that they can reduce the amount of browser oauth popups by installing keyring. It will indeed be shown if you don't install keyring and use the `OAuth2Authentication` class.

Maybe we can remove this and just write good docs. :)

Reply:
I understand the motiviation but it should use a logger and maybe not on warning. It's a hint, not an indication of a problem.

Reply:
OK, will use a logger. Not sure if anyone runs this with logging set to INFO, but I agree to your point that it's not really a warning from a logging perspective. Anyway it's explained in the docs already.